static/src/js/components/Map/ShapefileLayerExtended.js

```diff
@@ -0,0 +1,423 @@
+import L from 'leaflet'
+import forge from 'node-forge'
+import { isEqual } from 'lodash'
+
+import { eventEmitter } from '../../events/events'
+import { colorOptions } from '../SpatialSelection/SpatialSelection'
+import projections from '../../util/map/projections'
+
+const MAX_POLYGON_SIZE = 50
+
+const defaultOptions = {
+  selection: L.Draw.Polygon.prototype.options.shapeOptions
+}
+
+export class ShapefileLayerExtended extends L.Layer {
+  initialize(props) {
+    this.props = props
+    this.onAdd = this.onAdd.bind(this)
+    this.onRemovedFile = this.onRemovedFile.bind(this)
+    this.onSuccess = this.onSuccess.bind(this)
+    this.clickLayer = this.clickLayer.bind(this)
+    this.authToken = props.authToken
+    this.onMetricsMap = props.onMetricsMap
+    this.isProjectPage = props.isProjectPage
+    this.onToggleTooManyPointsModal = props.onToggleTooManyPointsModal
+    this.onChangeProjection = props.onChangeProjection
+
+    this.options = {
+      selection: L.extend({}, defaultOptions.selection, colorOptions)
+    }
+
+    this.selectedLayers = []
+
+    eventEmitter.on('shapefile.success', (file, resp) => {
+      this.onSuccess(file, resp)
+    })
+
+    eventEmitter.on('shapefile.removedfile', () => {
+      this.onRemovedFile()
+    })
+
+    const {
+      shapefile = {},
+      onFetchShapefile
+    } = props
+    const {
+      file,
+      shapefileId
+    } = shapefile
+
+    if (shapefileId && !file) {
+      onFetchShapefile(shapefileId)
+    }
+  }
+
+  drawNewShapefile(shapefile, selectedFeatures = []) {
+    this.onSuccess({}, shapefile, false, selectedFeatures)
+  }
+
+  onAdd(map) {
+    this.map = map
+
+    const { shapefile = {} } = this.props
+    const {
+      file,
+      selectedFeatures
+    } = shapefile
+
+    if (file) {
+      this.onSuccess({}, file, false, selectedFeatures)
+    }
+  }
+
+  onRemove() {
+    this.map = null
+    this.jsonLayer = null
+  }
+
+  onRemovedFile() {
+    if (this.jsonLayer != null) {
+      this.map.removeLayer(this.jsonLayer)
+    }
+
+    this.fileHash = null
+  }
+
+  // Leaflet 1.0+ changed the way that MultiPolygons are handled.
+  // Instead of creating a new layer for each polgon in a MultiPolygon
+  // feature (v0.7) it creates a single layer with multiple polygons.
+  // This means when you hover over one island in Hawaii it highlights
+  // all the islands, and passes all the polygons to _setConstraint.
+  // This method takes all the MultiPolygon geometries and separates them
+  // into individual polygons, to mimick the 0.7 functionality.
+  separateMultiPolygons(geojson) {
+    const featureIndexesToRemove = []
+    const { features } = geojson
+    const newFeaturesToAdd = []
+
+    features.forEach((feature, featureIndex) => {
+      const { geometry, properties = {} } = feature
```

Comment:
Main difference from HEAD is we were pull out properties here and using it throughout

Reply:
Yeah, we had to pull out the class into a new file to test it, which kind of hides the meaningful changes.

You can see those property additions on the original `ShapefileLayer.js` in an [older commit](https://github.com/nasa/earthdata-search/pull/1751/commits/5de8d6b4de827e132061ffca8011234b0b56403b).